src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Order/TreeItemOrderPropertyProvider.cs

```diff
@@ -64,6 +66,24 @@ private void ComputeIndices()
             }
         }
 
+        private bool TryGetIndex(IProjectTreeCustomizablePropertyContext propertyContext, out int index)
+        {
+            var didFindIndex = _displayOrderMap.TryGetValue(propertyContext.ItemName, out index);
+            if (didFindIndex)
+            {
+                return true;
+            }
+
+            var isLinked = propertyContext.Metadata.TryGetValue(LinkProperty, out var linkPath);
+            if (isLinked)
+            {
+                return _duplicateOrderMap.TryGetValue(linkPath, out index);
+            }
+            
+            return 
+                (propertyContext.Metadata.TryGetValue(FullPathProperty, out var fullPath) && _duplicateOrderMap.TryGetValue(fullPath.Replace(Path.GetDirectoryName(_project.FullPath) + PathSeparator, string.Empty), out index));
```

Comment:
can we format this to be on multiple lines please?

Reply:
This is trying to calculate what exactly? The ItemSpec? ie the value of Include in `<Compile Include="Foo.cs"/>` or the path relative to the project?

The item spec metadata will return the former, UnconfiguredProject.MakeRelative for the later.